ENH: Populate site content

kaitj opened the pull request:
PR adds in the building blocks for the web page. A lot of the content was pulled from the current site (khanlab.ca). There are still a few things to do before merging and going live, including:

- [x] Populating / replacing "news" placeholder
- [x] User profile placeholders (both for new users and repopulating with previous user profiles)
- [x] Updating any necessary pages with content
- [x] Test github actions for pulling publications _(seems to work as expected for non-default / `master` branch)_
- [x] Mobile version of site should also be tested.

kaitj commented:
@akhanf Made a few updates, but here's the code base! There should be some information where all the necessary files are / what can be updated if changes are needed in the `README`. I've added a list of things to the PR comment that should probably be updated before merging (which will make the site live). Also updated the publications csv file so it should be recent as of this morning! 

The github actions should work, though the `create_pull_request` action wasn't working properly on the dev branch. Not sure if it needs to be run from master branch (which was original intention), so that will need to be tested again once merged.

At the moment, any changes can only be previewed locally. It should be possible to set up a netlify deploy to preview any PRs before merging though I haven't looked into this (this was set up for some of the carpentries-incubator intro to neuroimaging lessons).

akhanf commented:
This is great, thanks again for putting this together! 

I'm trying to preview the website locally -- I cloned it and opened the index.html in firefox/chrome but I just see:

<img width="1318" alt="image" src="https://user-images.githubusercontent.com/11492701/133626352-e7481892-70a3-4083-a1f9-bb5846a6fb4c.png">

I guess I am missing something?

akhanf commented:
Ah I found the link to instructions in the readme now! nvm!

akhanf commented:
I added placeholder profile pages for (I think) everybody -- will leave it up to individuals to add in their own details instead of trying to copy from previous site..

akhanf commented:
ok I think we have things just about ready for the pre-launch (to khanlab.github.io at least, to have students then add in their content) -- I noticed only first 3 publications are showing up and no way to see more -- is that the typical behaviour?

akhanf commented:
OK 3-4 is probably fine for the main page, but how hard would it be to also
have a separate publications page that lists all of them (then can link to
that page from the menu)?

On Wed, Sep 22, 2021 at 11:00 AM Jason Kai ***@***.***> wrote:

> Yep, currently its show the latest 3 publications. Can change this by
> updating the code in the publications.html
> <https://github.com/khanlab/khanlab.github.io/blob/dev/_includes/index/publications.html>
> .
>
> {% assign publications = site.data.publications.publications %} {% for post in
> publications limit: 3 %}
>
> Can set the publications limit to any number - it just reads the N most
> recent ones from the csv file.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/khanlab/khanlab.github.io/pull/1#issuecomment-925014773>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACXV2XMNQV735DBNBY5XFDTUDHVRLANCNFSM5ECXR7ZA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>


-- 
*Ali R. Khan, PhD*
Canada Research Chair in Computational Neuroimaging
Assistant Professor, Medical Biophysics
Scientist, Robarts Research Institute
Schulich School of Medicine & Dentistry, Western University
e. ***@***.***
t. 519.931.5777 (x24280)
w. www.khanlab.ca

akhanf commented:
Ah actually I see how I might be able to do this -- will give it a try..

On Wed, Sep 22, 2021 at 11:04 AM Ali Khan ***@***.***> wrote:

> OK 3-4 is probably fine for the main page, but how hard would it be to
> also have a separate publications page that lists all of them (then can
> link to that page from the menu)?
>
> On Wed, Sep 22, 2021 at 11:00 AM Jason Kai ***@***.***>
> wrote:
>
>> Yep, currently its show the latest 3 publications. Can change this by
>> updating the code in the publications.html
>> <https://github.com/khanlab/khanlab.github.io/blob/dev/_includes/index/publications.html>
>> .
>>
>> {% assign publications = site.data.publications.publications %} {% for post in
>> publications limit: 3 %}
>>
>> Can set the publications limit to any number - it just reads the N most
>> recent ones from the csv file.
>>
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/khanlab/khanlab.github.io/pull/1#issuecomment-925014773>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/ACXV2XMNQV735DBNBY5XFDTUDHVRLANCNFSM5ECXR7ZA>
>> .
>> Triage notifications on the go with GitHub Mobile for iOS
>> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
>> or Android
>> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>>
>>
>
>
> --
> *Ali R. Khan, PhD*
> Canada Research Chair in Computational Neuroimaging
> Assistant Professor, Medical Biophysics
> Scientist, Robarts Research Institute
> Schulich School of Medicine & Dentistry, Western University
> e. ***@***.***
> t. 519.931.5777 (x24280)
> w. www.khanlab.ca
>


-- 
*Ali R. Khan, PhD*
Canada Research Chair in Computational Neuroimaging
Assistant Professor, Medical Biophysics
Scientist, Robarts Research Institute
Schulich School of Medicine & Dentistry, Western University
e. ***@***.***
t. 519.931.5777 (x24280)
w. www.khanlab.ca

kaitj commented:
I think I just requested access for netlify to this repo - should enable previews for PRs so `jekyll` and `ruby` don't necessarily have to be setup to see changes (limited to 300 minutes of build time per month).

> OK 3-4 is probably fine for the main page, but how hard would it be to also
have a separate publications page that lists all of them (then can link to
that page from the menu)?
> Ah actually I see how I might be able to do this -- will give it a try..

Let me know if you run into any issues, happy to throw a page together and add it to the navigation!

kaitj commented:
Made a couple of quick changes! 

* Noticed the image in the news popups were quite large, scaled these down so it wouldn't take up the whole page
* Added a logo for the CRC funding that was added, following their visual guidelines
* Fixed spacing around the funding logos from the addition of a new logo.

I will also set this PR ready to review so that we can merge it in for a prelaunch!

akhanf commented:
awesome, new changes look great, thanks! 
I've added the separate pubs page, adapting the existing code. It seems to render ok, but I do get a warning when it builds:

```    
Liquid Warning: Liquid syntax error (line 15): Expected end_of_string but found pipe in "authors | size > 5" in pubs.html
```
Any help would be appreciated!

kaitj commented:
First time I've seen that warning! But interestingly, it seems to be a combination of the pages' front matter with the Liquid syntax that seemed to be resulting in the error. The solution was to replace `authors | size > 5` with `authors.size > 5` for this page.

Alternative was to remove the front matter, but that would have also removed the css for the page.

akhanf commented:
thanks! I am going to merge this now so we can show it off in the lab meeting!
